xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/edit/editableProperty.js

```diff
@@ -130,7 +130,14 @@ define('editableProperty', ['jquery', 'xwiki-meta'], function($, xcontext) {
       // Allow others to enhance the editor.
       $(document).trigger('xwiki:dom:updated', {'elements': editor.toArray()});
       // Focus the first visible input.
-      editor.find(':input').filter(':visible').focus();
+      var editInput = editor.find(':input').filter(':visible');
+      editInput.focus();
+      // Make sure the edit input has an ID, and use the name of the input as a fallback
+      if (!editInput.attr('id')) {
+        editInput.attr('id', editInput.attr('name'));
+      }
+      // Bind the label to the newly generated edit input
+      editableProperty.find('label').attr('for',editInput.attr('id'));
```

Comment:
> Shouldn't this whole code be wrapped in an if-statement if there is an editInput?

Since this only happens when the promise to retrieve the edit display of a field is achieved, I didn't expect this to happen. I added a condition to finish the function early and skip those changes though, better safe than sorry :+1: 

> Also, this change seems to be much more general than AWM. I'm not saying it's bad, I'm just saying that it is more general.

Yup, I was navigating through multiple kinds of UI for objects and there was an issue with AWM inline edition. I tracked the cause and ended up changing things here. 

> Shouldn't there be code to remove the for attribute again from the label when editing is finished?

Technically it doesn't hurt to leave it. However it's better for HTML clarity to clean up the attribute when it's not useful anymore. Added a couple lines to clean this up on saving or canceling the edition